### DISCUSSION TITLE
Is it possible to restart a dynamic shovel via the HTTP API ?

### DISCUSSION THREAD

**Discussion by fgheysels** [rabbitmq/rabbitmq-server#2835] [Ideas] [answered]
We've an issue where we've defined a dynamic shovel on a device (actually a vessel) which has an unreliable / shaky connection.
Often, the shovel stops working because it was unable to connect for x times in a certain period of time.

We were thinking of creating a service which regularly checks the status of the shovel and restart the shovel if the shovel is down.  We'd to this via the HTTP API.
I've found [this](https://github.com/rabbitmq/rabbitmq-shovel-management/pull/34) PR which makes this functionality available, but I cannot find anything regarding getting the status of a shovel or restarting a shovel via the HTTP API.

Could you please extend the documentation with information on how to check the status of a shovel / restart a shovel via the REST API ?
Next to that, it would also be nice if the documentation could be enriched by specifying some details on how many connection attempts in which period of time need to happen before the shovel is terminated.

**Comment by michaelklishin** [comment]
I will convert this issue to a GitHub discussion. Currently GitHub will [automatically close and lock the issue](https://github.com/github/feedback/discussions/3197) even though your question will be transferred and responded to elsewhere. This is to let you know that we do not intend to ignore this but this is how the current GitHub conversion mechanism makes it seem for the users :(

**Comment by michaelklishin** [comment, marked as answer]
Moved to https://github.com/rabbitmq/rabbitmq-website/issues/1152. The site is open source so you are welcome to contribute to the [dynamic Shovels guide](https://github.com/rabbitmq/rabbitmq-website/blob/live/site/shovel-dynamic.md). Please add a new section similarly to Declaring a Shovel and Deleting a Shovel.

**Reply by fgheysels** [reply]
I was just having a look on how I could do this, but I saw that you've already done it. Thx!

**Comment by fgheysels** [comment]
I could determine how to restart a shovel by looking at the request that is launched via the shovel admin plugin.  (Sending a DELETE request to ´api/shovels/vhost/{hosname}/{shovelname}/restart´

However, it would also be nice if this is mentionned in the API docs.

**Reply by michaelklishin** [reply]
Done
